functional_testing/G8os/testsuite/c_advanced/tests_03_networking.py

```diff
@@ -0,0 +1,233 @@
+        r = c1_client.bash('ping -w5 {}'.format(c2_ip)).get()
+        self.assertEqual(r.state, 'SUCCESS', r.stdout)
+
+        self.lg('Check if (c3) can reach (c1), shouldn\'t be reachable')
+        r = c3_client.bash('ping -w5 {}'.format(c1_ip)).get()
+        self.assertEqual(r.state, 'ERROR', r.stdout)
+
+        self.lg('Delete the vxlan bridge (vx1), should succeed')
+        vxbridge = 'vxlbr' + vx1_id
+        self.ovscl.json('ovs.bridge-del', {"bridge": vxbridge})
+
+        self.lg('Check if (c1) can reach (c2), shouldn\'t be reachable')
+        r = c1_client.bash('ping -w5 {}'.format(c2_ip)).get()
+        self.assertEqual(r.state, 'ERROR', r.stdout)
+
+        self.lg('Terminate all machines, should succeed')
+        self.client.container.terminate(c1)
+        self.client.container.terminate(c2)
+        self.client.container.terminate(c3)
+
+        self.lg('{} ENDED'.format(self._testID))
+
+    def test002_vlans_connections(self):
+        """ g8os-030
+        *Test case for testing vlans connections*
+
+        **Test Scenario:**
+        #. Create dhcp server on a container
+        #. Create container (c1) on a new vlan bridge (v1), should succeed.
+        #. Create container (c2) connected on (v1) and connect it to default network.
+        #. Create conatiner (c3) on a new vlan bridge (v2)
+        #. Check that (c2) can reach the internet while (c1) can't.
+        #. Check if (c1) can reach (c2), should be reachable
+        #. Check if (c3) can reach (c1), shouldn't be reachable
+        #. Delete the vlan bridge, should succeed
+        #. Check if (c1) can reach (c2), shouldn't be reachable
+        #. Terminate all machines, should succeed
+        """
+
+        self.lg('{} STARTED'.format(self._testID))
+
+        self.lg('Create dhcp server on a container')
+        v1_id = str(randint(1, 4094))
+        dhcp_ip = '192.168.1.1'
+        nic = [{'type': 'default'}, {'type': 'vlan', 'id': v1_id, 'config': {'cidr': '{}/24'.format(dhcp_ip)}}]
+
+        dhcp_c = self.client.container.create(root_url=self.root_url, storage=self.storage, nics=nic)
+        dhcp_c_client = self.client.container.client(dhcp_c)
+        rs = dhcp_c_client.system('apt-get update').get()
+        self.assertEqual(rs.state, 'SUCCESS')
+        rs = dhcp_c_client.system('apt-get install -y dnsmasq-base').get()
+        self.assertEqual(rs.state, 'SUCCESS')
+        dhcp_c_client.system('dnsmasq --no-hosts --keep-in-foreground --listen-address=192.168.1.1 --interface=eth0 --dhcp-range=192.168.1.2,192.168.1.3,255.255.0.0 --bind-interfaces --except-interface=lo')
+        time.sleep(40)
+
+        self.lg('Create container (c1) on a new vlan bridge (v1), should succeed')
+        nic1 = [{'type': 'vlan', 'id': v1_id, 'config': {'dhcp': True}}]
+        c1 = self.client.container.create(root_url=self.root_url, storage=self.storage, nics=nic1)
+        c1_client = self.client.container.client(c1)
+        time.sleep(6)
```

Comment:
just making sure the container will take ip from the dhcp server as it sometimes took time